op-program/client/l2/engineapi/l2_engine_api.go

```diff
@@ -355,19 +358,20 @@ func (ea *L2EngineAPI) NewPayloadV3(ctx context.Context, params *eth.ExecutionPa
 		return &eth.PayloadStatusV1{Status: eth.ExecutionInvalid}, engine.InvalidParams.With(errors.New("nil parentBeaconBlockRoot post-cancun"))
 	}
 
-	if !ea.config().IsCancun(new(big.Int).SetUint64(uint64(params.BlockNumber)), uint64(params.Timestamp)) {
+	cfg := ea.config()
+
+	if !cfg.IsCancun(new(big.Int).SetUint64(uint64(params.BlockNumber)), uint64(params.Timestamp)) {
 		return &eth.PayloadStatusV1{Status: eth.ExecutionInvalid}, engine.UnsupportedFork.With(errors.New("newPayloadV3 called pre-cancun"))
 	}
 
-	// Payload must have eip-1559 params in ExtraData after Holocene
-	if ea.config().IsHolocene(uint64(params.Timestamp)) {
-		if err := eip1559.ValidateHoloceneExtraData(params.ExtraData); err != nil {
-			return &eth.PayloadStatusV1{Status: eth.ExecutionInvalid}, engine.UnsupportedFork.With(errors.New("invalid holocene extraData post-holocene"))
+	if cfg.IsOptimism() {
```

Comment:
nit: `cfg.IsOptimism()` is a no-op. We're always running op rules inside the program.

Reply:
[afe7721](https://github.com/ethereum-optimism/optimism/pull/17307/commits/afe77210967ab498d375953edcfb22184b3377d7)